src/LanguageServer.php

```diff
@@ -289,10 +324,48 @@ public function initialize(ClientCapabilities $capabilities, string $rootPath =
         });
     }
 
+    /**
+     * The initialized notification is sent from the client to the server after
+     * the client received the result of the initialize request but before the
+     * client is sending any other request or notification to the server.
+     *
+     * @return Promise
+     */
+    public function initialized(): Promise
+    {
+        return coroutine(function () {
+            if (!$this->rootPath) {
+                return;
+            }
+
+            // request configuration if it is supported
+            // support comes with protocol version 3.6.0
+            if ($this->clientCapabilities->workspace->configuration) {
+                $configurationitem = new ConfigurationItem();
+                $configurationitem->section = 'php';
+                $configuration = yield $this->client->workspace->configuration([$configurationitem]);
+                $options = $this->mapper->map($configuration[0], new Options());
+            }
+
+            // depending on the implementation of the client
+            // the workspace/didChangeConfiguration can be invoked before
+            // the response from the workspace/configuration request is resolved
+            if ($this->indexer->isIndexing()) {
+                return;
```

Comment:
And why do you need to return in that case?

Reply:
https://github.com/felixfbecker/php-language-server/pull/308#issuecomment-417713457

Because when the settings are different from default, the `didChangeConfiguration` is initiating the indexer. When it is already running, we know the running indexer is using the latest settings.